datafusion/core/src/execution/context.rs

```diff
@@ -1833,6 +1841,28 @@ impl From<&SessionState> for TaskContext {
     }
 }
 
+fn create_dialect_from_str(dialect_name: &str) -> Result<Box<dyn Dialect>> {
```

Comment:
I have ported this code upstream here: https://github.com/sqlparser-rs/sqlparser-rs/pull/848

```suggestion
// TODO: remove when https://github.com/sqlparser-rs/sqlparser-rs/pull/848 is released
fn create_dialect_from_str(dialect_name: &str) -> Result<Box<dyn Dialect>> {
```